### DISCUSSION TITLE
vercel in GitLab CI: Can't pass env vars as in documentation

### DISCUSSION THREAD

**Discussion by brycewray** [vercel/vercel#4730] [Help] [answered]
I am attempting to use GitLab CI with `vercel` (rather than the direct Vercel-to-GitLab connection) to build a static site while I test Vercel for the purposes of comparing it with my current site host, Netlify. In GitLab, I have set variables `VERCEL_TOKEN`, `VERCEL_ORG_ID`, and `VERCEL_PRODUCT_ID`, and want to provide them as environment variables. Based on the "Usage Example" at https://vercel.com/docs/cli#commands/overview/unique-options/env, I have the following line in my `.gitlab-ci.yml` file:

```
    vercel --token $VERCEL_TOKEN --env [VERCEL_ORG_ID=$VERCEL_ORG_ID VERCEL_PROJECT_ID=$VERCEL_PROJECT_ID]
```

...and, each time I try this, it errors out at that line with the following (I am purposely inserting "`<<$VERCEL_PROJECT_ID>>`" below to substitute for the actual value, which it returns correctly and in full):

```
 Vercel CLI 19.1.1
 Error! The specified file or directory "VERCEL_PROJECT_ID=<<$VERCEL_PROJECT_ID>>]" does not exist.
```
Note that the closing bracket is included in the quotation marks.

Is there something I'm missing about the example in the linked documentation? It says this should work:
```
vercel --env [key1=value1 key2=value2 key3=value3]
```

Thanks in advance for any help one can offer a Vercel noob.

**Comment by mcsdevv** [comment]
Hello Bryce, can you try this without the `[` and `]` characters? They are there to indicate that variable input is required.

**Comment by brycewray** [comment, marked as answer]
@mcsdevv Thanks for the suggestion. Same result, except no bracket inside the parentheses in the error message. My command line was:
```
vercel --token $VERCEL_TOKEN --env VERCEL_ORG_ID=$VERCEL_ORG_ID VERCEL_PROJECT_ID=$VERCEL_PROJECT_ID --prod
```

...and the error message was (again, substituting `<<$VERCEL_PROJECT_ID>>` for the actual variable return, which it sends back correctly):
```
Error! The specified file or directory "VERCEL_PROJECT_ID=<<$VERCEL_PROJECT_ID>>" does not exist.
 ERROR: Job failed: exit code 1
```

Do I need to wrap those vars in something else besides brackets so it will recognize the second one, too? Or, alternatively, do I need to have `--env` in front of each, separately?

**Reply by paulogdm** [reply]
Usually when users are trying to use `VERCEL_ORG_ID` they do so with:
```
VERCEL_ORG_ID=$VERCEL_ORG_ID VERCEL_PROJECT_ID=$VERCEL_PROJECT_ID vercel --token $VERCEL_TOKEN  --prod
```

**Reply by brycewray** [reply]
@paulogdm Thanks very much! That worked perfectly.
